C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackBotWorker.cs

```diff
@@ -19,7 +28,20 @@ public class SlackBotWorker : BotWorker
         /// <param name="config">Normally, a DialogContext object.  Can also be the id of a team.</param>
         public SlackBotWorker(Botkit botkit, object config) : base(botkit, config)
```

Comment:
This constructor suggests that the `config` parameter can be both a general object or a string. I propose that we can:

1. Create two constructors, one for the string _config_ and the second one for the _object_ type (which we should define). For the string constructor, when we call the _base_ constructor, we can use a static method for creating the _config_ parameter with the provided string
1. Only provide a constructor whose _config_ parameter is a specific object type.

_**NOTE**_: Regarding the _object config_ I think that obe possible type is the `DialogContext` ([link](https://github.com/southworkscom/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs)), because, as it's said in the parent constructor ([link](https://github.com/southworkscom/botkit/blob/master/packages/botkit/src/botworker.ts#L23)), the config object is ```An object typically containing { dialogContext, reference, context, activity }```, and _DialogContext_ accomplish almost all these requirements (except reference)

Reply:
I think that there is a misunderstanding in the usage of the _base_ constructor. Note that the _base constructor_ will be called first an then the remaining code will be executed. After reading this code again, it seems that you create the _**config**_ object for using it in the base constructor, but this doesn't work this way. Check this [link](https://www.dotnetperls.com/base) which explains the workflow of the constructors when using the base one.

Reply:
This is going to be applied in another PR/branch